component/kafka/simple/component.go

```diff
@@ -0,0 +1,359 @@
+	brokers                  []string
+	saramaConfig             *sarama.Config
+	proc                     kafka.BatchProcessorFunc
+	failStrategy             kafka.FailStrategy
+	batchSize                uint
+	batchTimeout             time.Duration
+	retries                  uint
+	retryWait                time.Duration
+	partitionConsumerFactory func(ctx context.Context) (sarama.Client, sarama.Consumer, map[int32]sarama.PartitionConsumer, error)
+	// DurationOffset options
+	durationBasedConsumer bool
+	durationOffset        time.Duration
+	timeExtractor         func(*sarama.ConsumerMessage) (time.Time, error)
+	// NotificationOnceReachingLatestOffset options
+	latestOffsetReachedChan chan<- struct{}
+	startingOffsets         map[int32]int64
+	latestOffsets           map[int32]int64
+	once                    sync.Once
+}
+
+// New initializes a new simple kafka consumer component with support for functional configuration.
+func New(name string, brokers []string, topic string, proc kafka.BatchProcessorFunc, saramaCfg *sarama.Config, oo ...OptionFunc) (*Component, error) {
+	var errs []error
+	if name == "" {
+		errs = append(errs, errors.New("name is required"))
+	}
+
+	if validation.IsStringSliceEmpty(brokers) {
+		errs = append(errs, errors.New("brokers are empty or have an empty value"))
+	}
+
+	if topic == "" {
+		errs = append(errs, errors.New("topic is required"))
+	}
+
+	if proc == nil {
+		errs = append(errs, errors.New("work processor is required"))
+	}
+
+	if len(errs) > 0 {
+		return nil, patronErrors.Aggregate(errs...)
+	}
+
+	cmp := &Component{
+		name:         name,
+		brokers:      brokers,
+		topic:        topic,
+		proc:         proc,
+		retries:      defaultRetries,
+		retryWait:    defaultRetryWait,
+		batchSize:    defaultBatchSize,
+		batchTimeout: defaultBatchTimeout,
+		failStrategy: defaultFailureStrategy,
+		saramaConfig: saramaCfg,
+	}
+
+	cmp.partitionConsumerFactory = cmp.createPartitionConsumers
+
+	for _, optionFunc := range oo {
+		err := optionFunc(cmp)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return cmp, nil
+}
+
+// Run starts the consumer processing loop to process messages from Kafka.
+func (c *Component) Run(ctx context.Context) error {
+	retries := int(c.retries)
+
+	if retries == 0 {
+		return c.runWithoutRetry(ctx)
+	}
+	return c.runWithRetry(ctx, retries)
+}
+
+func (c *Component) runWithoutRetry(ctx context.Context) error {
+	_, err := c.processing(ctx)
+	if err != nil {
+		return fmt.Errorf("simple consumer error on topic %s: %w", c.topic, err)
+	}
+	return nil
+}
+
+func (c *Component) runWithRetry(ctx context.Context, retries int) error {
+	var hasProcessedMessages bool
+	var err error
+
+	for i := 0; ; i++ {
+		hasProcessedMessages, err = c.processing(ctx)
+		if err != nil {
+			log.Errorf("simple consumer error on topic %s: %v", c.topic, err)
+		}
+
+		if isContextDone(ctx) {
+			break
+		}
+		if hasProcessedMessages {
+			i = 0
+		}
+		if i >= retries {
+			break
+		}
```

Comment:
I think It would be helpful to add a log here indicating that we have run out of retries